cmd/argocd/commands/cluster.go

```diff
@@ -40,6 +39,24 @@ func NewClusterCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clientc
 	return command
 }
 
+// DefaultKubeConfigDir returns the local configuration path for settings such as cached authentication tokens.
+func DefaultKubeConfigDir() (string, error) {
```

Comment:
We shouldn't invent our own method for resolving a kubeconfig. This method will not honor the --kubeconfig flag. The previous logic to get a kubeconfig contents using `config, err := configAccess.GetStartingConfig()` needs to be preserved.

Reply:
@jessesuen have now restored that code.